### DISCUSSION TITLE
Missing libfaustcsound.so opcode library in plugins repository in the csound github project

### DISCUSSION THREAD

**Discussion by lorendir** [community/community#21444] [Other Feature Feedback, Questions, & Ideas]
<p>Hello,</p>
<p>As I am working on a computer music instruments project, I am missing the libfaustcsound.so opcode library which, as I was told, is in the plugins repository of the csound GitHub project. But, unless I missed it, I couldn’t find it in the GitHub csound-plugins repository, nor anywhere else.</p>
<p>I would therefore very much appreciate any hint or advice from anyone with (or without) some familiarity with csound and faust on how to get this file.</p>
<p>With my best regards in the meantime.</p>

**Comment by tajmone** [comment]
<p>Please, always provide links to the projects you mention, which spares those trying to look into the problem from having to search for them on your behalf.</p>
<ul>
<li><a href="https://github.com/csound/csound" class="inline-onebox" rel="noopener nofollow ugc">GitHub - csound/csound: Main repository for Csound</a></li>
<li><a href="https://csound.com" rel="noopener nofollow ugc">https://csound.com</a></li>
</ul>
<p>Are the above links the project you are speaking about?</p>
<aside class="quote no-group" data-username="lorendir" data-post="1" data-topic="196812">
<div class="title">
<div class="quote-controls"></div>
 lorendir:</div>
<blockquote>
<p>I was told, is in the plugins repository of the csound GitHub project.</p>
</blockquote>
</aside>
<p>There doesn’t seem to be a <code>plugins</code> folder in the repository, nor any file with basename <code>libfaustcsound</code>.</p>
<p>The closest match I could find is in this CMake file:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/csound/csound/blob/c7601d11f029f24590917d4a1797e247eafdef21/cmake/Modules/FindFAUST.cmake">
  <header class="source">

      <a href="https://github.com/csound/csound/blob/c7601d11f029f24590917d4a1797e247eafdef21/cmake/Modules/FindFAUST.cmake" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/csound/csound/blob/c7601d11f029f24590917d4a1797e247eafdef21/cmake/Modules/FindFAUST.cmake" target="_blank" rel="noopener nofollow ugc">csound/csound/blob/c7601d11f029f24590917d4a1797e247eafdef21/cmake/Modules/FindFAUST.cmake</a></h4>


    <pre><code class="lang-cmake"># Find Faust2

include(FindPackageHandleStandardArgs)

find_path(
  FAUST_INCLUDE_DIR faust/dsp/llvm-dsp.h
  HINTS
  /opt/lib/faust/architecture/
  /usr/lib/faust/architecture/
  /usr/local/lib/faust/architecture/
  "${FAUST_INCLUDE_DIR_HINT}"
  )

set(FAUST_NAMES ${FAUST_NAMES} libfaust.so libfaust.dylib faust.dll faust libfaust)
find_library(FAUST_LIBRARY
  NAMES ${FAUST_NAMES}
  HINTS "${FAUST_LIB_DIR_HINT}")


find_package_handle_standard_args(FAUST FAUST_INCLUDE_DIR FAUST_LIBRARY)
</code></pre>


  This file has been truncated. <a href="https://github.com/csound/csound/blob/c7601d11f029f24590917d4a1797e247eafdef21/cmake/Modules/FindFAUST.cmake" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>which would produce a <code>libfaust.so</code> dynamic library. Is that what you need?</p>
<p>In any case, files with the <code>.so</code> extension are precompiled dynamic libraries, which are usually ignored from repositories and users are expected to compile them themselves. You should be able to extract that file from a distribution package, since dynamic libraries are shipped with the application; so you can look inside the application folders (possibly in <code>modules/</code> or <code>plugins/</code>) and just copy it — beware that it will only work if its version matches the one required by the application using it.</p>

**Comment by lorendir** [comment, marked as answer]
<p>Thank you very much for your reply. Please forgive me for not having given the link to the csound repository in GitHub. Yes, indeed, this is exactly what I was looking for and I very much appreciate your having taken the time to explain how to extract precompiled .so files. I have not tried to compile the libfaust.so file yet but it shouldn’t be a problem, so I believe this issue can be considered as solved and can therefore be closed.</p>
<p>Thank you again for your assistance.</p>

**Comment by tajmone** [comment]
<p>You’re most welcome.</p>
